classes/services/PKPContextService.inc.php

```diff
@@ -299,6 +299,43 @@ public function validate($action, $props, $allowedLocales, $primaryLocale) {
 			}
 		});
 
+		// Ensure that the form and submission locales are available for the context
+		$validator->after(function($validator) use ($action, $props) {
+			$context = Application::get()->getRequest()->getContext();
+			if (isset($props['id']) && $context->getId() !== $props['id']) {
+				$context = $this->get($props['id']);
+			}
+
+			$getCurrentPropValue = function($prop) use ($props, $context) {
+				if (isset($props[$prop])) {
+					return $props[$prop];
+				}
+
+				return $context->getData($prop);
+			};
+
+			$supportedLocales = $getCurrentPropValue('supportedLocales');
+			$supportedFormLocales = $getCurrentPropValue('supportedFormLocales');
+			$supportedSubmissionLocales = $getCurrentPropValue('supportedSubmissionLocales');
+
+			$unsupportedFormLocales = array_diff($supportedFormLocales, $supportedLocales);
+			$unsupportedSubmissionLocales = array_diff($supportedSubmissionLocales, $supportedLocales);
```

Comment:
For readability, I'd suggest moving this down a little closer to where it's used.